bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/GarbageCollectorThread.java

```diff
@@ -512,10 +512,16 @@ protected void compactEntryLog(EntryLogMetadata entryLogMeta) {
             // indicates that compaction is in progress for this EntryLogId.
             return;
         }
-        // Do the actual compaction
-        compactor.compact(entryLogMeta);
-        // Mark compaction done
-        compacting.set(false);
+
+        try {
+            // Do the actual compaction
+            compactor.compact(entryLogMeta);
+        } catch (Exception e) {
```

Comment:
Do we need to catch *Exception*
Isn't it enough to use try/finally to fix you problem?

Reply:
If we don't catch the exception here, I think that the following for loop is interrupted at that time.
https://github.com/apache/bookkeeper/blob/b2e099bbc7b13f13825fe78ab009ca132cb3a9ba/bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/GarbageCollectorThread.java#L441-L465
In our case, there is a problem with a particular entry log, so this loop will always stop when `GarbageCollectorThread` try to compact that log. Because of that, I think that the subsequent entry logs are not compacted every time.